### PR TITLE
Add GetSuiteRates() function

### LINKED ISSUE #208: Extend .combineFRatesMatrices to allow accessing all types of rateMatrix
After discussing with @rhanIDM about #204 
We think the functionality that researchers need is something similar to .combineFRatesMatrices (we should make this a public function)
which should take an argument of "Category" such as maleMortality, femaleFertility etc and return a data frame wth columns:
['Label', 'Year', 'Rate', 'Trial']

### PR DESCRIPTION
**Pull request by celiot-IDM**
Fixes #208

### CODE REVIEW

**Comment by celiot-IDM** [comment]
@MeWu-IDM : this one also worked! Thank you.

File: pacehrh/R/pace_utils.R [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+    warning(paste0("Invalid population rate category (", rateCategory, ")"))
+    return(NULL)
+  }
+
+  trials <- names(results)
+
+  l <- lapply(trials, function(trial) {
+    t <-
+      results[[trial]][["PopulationRates"]][[rateCategory]][["ratesMatrix"]]
+    t <- as.data.table(t, keep.rownames = "Label")
```
Comment (MeWu-IDM) [inline pacehrh/R/pace_utils.R:33]: The model_input.xlsx seems to have missing labels, we need to agree to add this field as required (currently no validation rules are set for this https://institutefordiseasemodeling.github.io/PACE-HRH/articles/input-validation.html#sheet-popvalues

File: pacehrh/tests/testthat/test-pace_utils.R [hunk 2]
```diff
@@ -10,3 +10,37 @@ test_that("Utilities: is.blank", {
...
+
+  df <- pacehrh::GetSuiteRates(results, rateCategory = "femaleFertility")
+  testthat::expect_true(!is.null(df))
+
+  cols <- c("Label",
+            "Trial",
+            "Year",
+            "Rate")
+
+  testthat::expect_equal(names(df), cols)
```
Comment (MeWu-IDM) [inline pacehrh/tests/testthat/test-pace_utils.R:45]: suggest to add value sanity check to be thorough 
such as 
```
for (i in unique(df$Label[!is.na(df$Label)])) {
  expect_equal(
    sum(results[[1]]$PopulationRates[[rateCategory]]$ratesMatrix[i,]),  
    as.vector(df %>% filter(Label==i ) %>% filter(Trial ==1)  %>%summarise(sum(Rate)))[[1]]
  ) 
}
```

**Review by MeWu-IDM — COMMENTED** [review]
I think the calculation is correct but we need to coordinate with @rhanIDM to make sure "label" is required otherwise analysis may get bad results using GetSuiteRates with bunhc of NAs

File: pacehrh/R/pace_utils.R (hunk 1 above)
Comment (celiot-IDM) [reply pacehrh/R/pace_utils.R:33]: Zero rates are both valid (fertility rates above/below fertile range), and the blank label is a signal to not show them in reports.

**Review by MeWu-IDM — APPROVED** [review]
will address usage and documentation in https://github.com/InstituteforDiseaseModeling/PACE-HRH/issues/212

**Pull request merged**